scripts/linters/js_ts_linter.py

```diff
@@ -210,21 +209,20 @@ def _get_expressions_from_parsed_script(self):
 
         return parsed_expressions_in_files
 
-    def _compile_ts_file(self, filepath, dir_path):
-        """Compiles a typescript file and returns the path for compiled
-        js file.
-        """
-        cmd = (
-            './node_modules/typescript/bin/tsc -outDir %s -allowJS %s '
-            '-lib %s -noImplicitUseStrict %s -skipLibCheck '
-            '%s -target %s -typeRoots %s %s typings/*') % (
-                dir_path, 'true', 'es2017,dom', 'true',
-                'true', 'es5', './node_modules/@types', filepath)
-        subprocess.call(cmd, shell=True, stdout=subprocess.PIPE)
+    def _get_compiled_ts_filepath(self, filepath):
+        """Returns the path for compiled ts file."""
         compiled_js_filepath = os.path.join(
-            dir_path, os.path.basename(filepath).replace('.ts', '.js'))
+            os.getcwd(),
+            'tmpcompiledjs/',
+            os.path.relpath(filepath).replace('.ts', '.js'))
         return compiled_js_filepath
 
+    def _compile_ts(self):
```

Comment:
_compile_all_ts_files perhaps?

Incidentally, how long does this take?

Also, maybe this is a good place to explain why we do it. You could say "Previously, we only compiled the TS files that were needed, but this didn't work because XXX. For more details, please see PR #YYY." or something like that.

(I'm also wondering if the "correct" fix is to fix the issue with relative paths, given that that's what you've found was the cause of the issue. I'll defer to @Hudda and @vojtechjelinek for a decision on that.)

Reply:
As @ankita240796 mentioned in #9458 it might be unnecessary to do this, we should see how many TS files can be deconstructed by `esprima` directly and don't need to be recompiled to JavaScript.

Reply:
see reply here for benchmarks: https://github.com/oppia/oppia/issues/9458#issuecomment-646764064